kbl2vec/src/main/java/com/foursoft/harness/kbl2vec/transform/placements/OnPointPlacementTransformer.java

```diff
@@ -25,28 +25,67 @@
  */
 package com.foursoft.harness.kbl2vec.transform.placements;
 
-import com.foursoft.harness.kbl.v25.KblFixingAssignment;
-import com.foursoft.harness.kbl2vec.core.Query;
-import com.foursoft.harness.kbl2vec.core.TransformationContext;
-import com.foursoft.harness.kbl2vec.core.TransformationResult;
-import com.foursoft.harness.kbl2vec.core.Transformer;
+import com.foursoft.harness.kbl.common.HasIdentification;
+import com.foursoft.harness.kbl.v25.*;
+import com.foursoft.harness.kbl2vec.core.*;
+import com.foursoft.harness.vec.v2x.VecNodeLocation;
 import com.foursoft.harness.vec.v2x.VecOnPointPlacement;
 import com.foursoft.harness.vec.v2x.VecPlaceableElementRole;
 import com.foursoft.harness.vec.v2x.VecSegmentLocation;
 
-public class OnPointPlacementTransformer implements Transformer<KblFixingAssignment, VecOnPointPlacement> {
+import java.util.Comparator;
+import java.util.List;
+
+public class OnPointPlacementTransformer implements Transformer<ConnectionOrOccurrence, VecOnPointPlacement> {
 
     @Override
     public TransformationResult<VecOnPointPlacement> transform(final TransformationContext context,
-                                                               final KblFixingAssignment source) {
+                                                               final ConnectionOrOccurrence source) {
+        if (!(source instanceof FixedComponent || source instanceof LocatedComponent)) {
```

Comment:
Almost! 😉 If it is a `FixedComponent` or `LoccatedComponent` it should still return `noResult()` if there is no 
`locatedComponent.getRefNode()` and `fixing.getRefFixingAssignment()` otherwise you will create a `OnPointPlacement` without locations.

Like this:

<img width="502" height="172" alt="Image" src="https://github.com/user-attachments/assets/0319707e-ac66-4117-8874-88cbc460cf1a" />

Which is illegal:

<img width="135" height="222" alt="Image" src="https://github.com/user-attachments/assets/fd8069ae-8e94-48ae-b76b-c27da5ef3715" />